src/backup/on_demand/on_demand_should_backup_predicate.py

```diff
@@ -0,0 +1,22 @@
+import logging
+
+from src.backup.abstract_should_backup_predicate import \
+    AbstractShouldBackupPredicate
+
+
+class OnDemandShouldBackupPredicate(AbstractShouldBackupPredicate):
+
+    def test(self, big_query_table_metadata, table_entity):
+        if not self._is_possible_to_copy_table(big_query_table_metadata):
+            return False
+
+        if big_query_table_metadata.is_empty():
```

Comment:
Can we move this log to abstract method?

Reply:
done